ckanext/xloader/plugin.py

```diff
@@ -68,6 +76,23 @@ def configure(self, config_):
                     )
                 )
 
+    # IPipeValidation
+
+    def receive_validation_report(self, validation_report):
+        if utils.requires_successful_validation_report():
+            res_dict = toolkit.get_action('resource_show')({'ignore_auth': True},
+                                                           {'id': validation_report.get('resource_id')})
+            if (toolkit.asbool(toolkit.config.get('ckanext.xloader.validation.enforce_schema', True))
+                or res_dict.get('schema', None)) and validation_report.get('status') != 'success':
+                    # either validation.enforce_schema is turned on or it is off and there is a schema,
```

Comment:
Comment could be clearer, but okay.

I might phrase it more as "A schema is either present, or required".

Reply:
Inline comments updated for clarity and human readability (I don't even know what I was trying to say now haha.)